src/main/java/org/opensrp/repository/postgres/PlanRepositoryImpl.java

```diff
@@ -0,0 +1,213 @@
+package org.opensrp.repository.postgres;
+
+import org.apache.commons.lang.StringUtils;
+import org.opensrp.domain.PlanDefinition;
+import org.opensrp.domain.postgres.*;
+import org.opensrp.repository.PlanRepository;
+import org.opensrp.repository.postgres.mapper.PlanMetadataMapper;
+import org.opensrp.repository.postgres.mapper.custom.CustomPlanMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import static org.opensrp.util.Utils.isEmptyList;
+
+/**
+ * Created by Vincent Karuri on 02/05/2019
+ */
+
+@Repository
+public class PlanRepositoryImpl extends BaseRepositoryImpl<PlanDefinition> implements PlanRepository {
+
+    @Autowired
+    private CustomPlanMapper planMapper;
+
+    @Autowired
+    private PlanMetadataMapper planMetadataMapper;
+    
+    @Override
+    public PlanDefinition get(String id) {
+        if (StringUtils.isBlank(id)) {
+            return null;
+        }
+
+        PlanExample planExample = new PlanExample();
+        planExample.createCriteria().andIdEqualTo(id).andIsDeletedNotEqualTo(true);
+
+        List<Plan> pgPlan = planMapper.selectByExample(planExample);
+        List<PlanDefinition> plan = convert(pgPlan);
+
+        return isEmptyList(plan) ? null : plan.get(0);
+    }
+
+    @Override
+    public void add(PlanDefinition plan) {
+        if (getUniqueField(plan) == null) {
+            return;
+        }
+
+        if (retrievePrimaryKey(plan) != null) { 
+            return; // plan already added
+        }
+
+        Plan pgPlan = convert(plan);
+        if (pgPlan == null) {
+            return;
+        }
+        pgPlan.setIsDeleted(false);
```

Comment:
Consider tracking date deleted instead of boolean

Reply:
Done.